internal/transport/transport_test.go

```diff
@@ -476,10 +486,42 @@ func setUpWithNoPingServer(t *testing.T, copts ConnectOptions, connCh chan net.C
 		}
 		framer := http2.NewFramer(conn, conn)
 		if err := framer.WriteSettings(); err != nil {
-			t.Errorf("Error at server-side while writing settings: %v", err)
+			t.Errorf("Error at server-side while writing settings: %v", framer.ErrorDetail())
 			close(connCh)
 			return
 		}
+		if respModeCh != nil {
+			if err := framer.WriteSettingsAck(); err != nil {
+				t.Errorf("Error while writing settings: %v", err)
+				return
+			}
+			go func() {
```

Comment:
A goroutine in a goroutine?

Why not push to `connCh` then do this logic?

Reply:
@dfawley Could you please clarify your comment here and what your suggestion is? I'm not following what the issue is to use a goroutine in a goroutine, I'll appreciate your help on this part. Thanks